Compilation failure for qemu_x86 with CONFIG_DEBUG_INFO=y

For the  sample apps (e.g. samples/net/lwm2m_client), on defining CONFIG_DEBUG_INFO=y in project configuration, it leads to compilation failure with following log.

----------------------------------------------------------------------------------------------------------------
/home/b16394/network_protocols/networkprotocols/WP1/zephyr/arch/x86/core/swap.S: Assembler messages:
/home/b16394/network_protocols/networkprotocols/WP1/zephyr/arch/x86/core/swap.S:467: Error: no instruction mnemonic suffix given and no register operands; can't size instruction
make[5]: *** [/home/b16394/network_protocols/networkprotocols/WP1/zephyr/scripts/Makefile.build:260: arch/x86/core/swap.o] Error 1
make[4]: *** [/home/b16394/network_protocols/networkprotocols/WP1/zephyr/scripts/Makefile.build:357: arch/x86/core] Error 2
make[3]: *** [/home/b16394/network_protocols/networkprotocols/WP1/zephyr/scripts/Makefile.build:357: arch/x86] Error 2
make[2]: *** [/home/b16394/network_protocols/networkprotocols/WP1/zephyr/Makefile:1072: arch] Error 2
make[2]: Leaving directory '/home/b16394/network_protocols/networkprotocols/WP1/zephyr/samples/net/lwm2m_client/outdir/qemu_x86'
make[1]: *** [Makefile:178: sub-make] Error 2
make[1]: Leaving directory '/home/b16394/network_protocols/networkprotocols/WP1/zephyr'
make: *** [/home/b16394/network_protocols/networkprotocols/WP1/zephyr/Makefile.inc:82: all] Error 2
[b16394@lti lwm2m_client]$

--- a/arch/x86/core/swap.S
+++ b/arch/x86/core/swap.S
@@ -464,7 +464,7 @@ SECTION_FUNC(TEXT, _x86_thread_entry_wrapper)
 	pop	%ecx
 	push	$0	 /* Null return address */
 #elif defined(CONFIG_GDB_INFO) || defined(CONFIG_DEBUG_INFO)
-	mov	$0, (%esp) /* zero initialEFLAGS location */
+	movl	$0, (%esp) /* zero initialEFLAGS location */
 #endif
 	jmp	*%edi
 #endif

